blark/transform.py

```diff
@@ -614,30 +614,43 @@ class Variable(Expression):
     ...
 
 
+@dataclass
 @_rule_handler(
     "indirection_type",
     "pointer_type",
 )
-class IndirectionType(Enum):
+class IndirectionType(Literal):
     """Indirect access through a pointer or reference."""
-    none = enum.auto()
-    pointer = enum.auto()
-    reference = enum.auto()
+    pointer_depth: int
+    reference: bool
+    meta: Optional[Meta] = meta_field()
 
     @staticmethod
-    def from_lark(token: Optional[lark.Token]) -> IndirectionType:
-        return {
-            "NONE": IndirectionType.none,
-            "POINTER TO": IndirectionType.pointer,
-            "REFERENCE TO": IndirectionType.reference,
-        }[str(token).upper()]
+    def from_lark(*tokens: Tuple[lark.Token]) -> IndirectionType:
```

Comment:
`*args` annotations are `*args: type_of_individual_arg`. I think this should probably be this:

```suggestion
    def from_lark(*tokens: lark.Token) -> IndirectionType:
```

Reply:
of course! don't know why I didn't think of that....